apm-sniffer/apm-sdk-plugin/kafka-plugin/src/main/java/org/apache/skywalking/apm/plugin/kafka/KafkaProducerInterceptor.java

```diff
@@ -41,16 +42,16 @@
 
     @Override
     public void beforeMethod(EnhancedInstance objInst, Method method, Object[] allArguments, Class<?>[] argumentsTypes,
-        MethodInterceptResult result) throws Throwable {
+                             MethodInterceptResult result) throws Throwable {
 
         ContextCarrier contextCarrier = new ContextCarrier();
 
-        ProducerRecord record = (ProducerRecord)allArguments[0];
-        String topicName = (String)((EnhancedInstance)record).getSkyWalkingDynamicField();
-
-        AbstractSpan activeSpan = ContextManager.createExitSpan(OPERATE_NAME_PREFIX + topicName + PRODUCER_OPERATE_NAME_SUFFIX, contextCarrier, (String)objInst.getSkyWalkingDynamicField());
+        ProducerRecord record = (ProducerRecord) allArguments[0];
+        String topicName = record.topic();
+        String key = record.key() == null ? null : (String) record.key();
+        AbstractSpan activeSpan = ContextManager.createExitSpan(OPERATE_NAME_PREFIX + topicName + "/" + key + PRODUCER_OPERATE_NAME_SUFFIX, contextCarrier, (String) objInst.getSkyWalkingDynamicField());
```

Comment:
We don't need to add key into OperatorName. Because it will create too many span entry.

Reply:
Yes. Please remove it from the operation name. Operation name is a group, should not relate to business inform.

Reply:
Ok, I'm going to change that